cmd/namespaces/main.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/codegangsta/cli"
+	"github.com/docker/distribution/namespace"
+)
+
+// Important considerations:
+//	1. Why are aliases and namespace config in the same file? It's important
+//       to understand the mapping of namespaces to remotes. To keep this
+//       simple, the mapping to a remote is managed within the namespace file.
+//  2. Let's say we have the default config with foo.com trust added. We
+//     should still resolve the aliases to pick up extra config, even if we
+//     have a match:
+//
+//       $ ./namespaces list foo.com
+//       docker.com/    pull     https://registry.docker.com
+//       docker.com/    push     https://registry.docker.com
+//       foo.com/       trust    http://foo.com/ca
+//       *              alias    docker.com/library/
+//
+//     This is actually a bug.
+//  3. Let's take the case of where we want to push to production. Does that
+//     need a namespace? No.
+
+// Broken cases:
+//  1. 	./namespaces add local/* alias redhat.com/
+//         -> doesn't get added
+
+// silly globals for now.
+var discoverer hardCodedDiscoverer
+
+var namespaces = namespace.NewEntries()
+var defaults = namespace.NewEntries()
+
+func addDefault(scope, action string, args ...string) {
+	entry, err := namespace.NewEntry(scope, action, args...)
+	if err != nil {
+		panic(err)
+	}
+	if err := defaults.Add(entry); err != nil {
+		panic(err)
+	}
+}
+
+func init() {
+	addDefault("docker.com", "pull", "https://registry.docker.com")
+	addDefault("docker.com", "push", "https://registry.docker.com")
+}
+
+func main() {
+	if err := read(); err != nil {
+		log.Fatalf("error reading configuration: %v", err)
+	}
+	defer write()
+
+	app := cli.NewApp()
+	app.Name = "dist"
+	app.Usage = "dist tool demo"
+
+	app.Commands = []cli.Command{
+		commandAdd,
+		commandDiscover,
+		commandList,
+		commandRemove,
+	}
+
+	app.RunAndExitOnError()
+}
+
+func read() error {
+	fp, err := os.Open(".namespace.cfg")
```

Comment:
This should somehow be configurable. When this code gets merged into the engine the file will likely live as `~/.docker/namespace.cfg` and this will be the main tool for making manual changes to that file (the engine may just treat it as read only).